example/lib/waveforms_dashboard.dart

```diff
@@ -77,39 +78,36 @@ class _WaveformsDashboardState extends State<WaveformsDashboard> {
   }
 
   Future<void> playAudio() async {
-    await audioPlayer.load(audioData[1]);
-    await audioPlayer.play(audioData[1]);
+    await audioPlayer.setSourceAsset(audioData[1]);
+    await audioPlayer.resume();
     // maxDuration in milliseconds
     await Future.delayed(const Duration(milliseconds: 200));
 
-    int maxDurationInmilliseconds =
-        await audioPlayer.fixedPlayer!.getDuration();
+     await audioPlayer.getDuration()
+        .then((value) => maxDuration = value!);
```

Comment:
It's not safe to bang! a better approach would be to set a default duration to 0.

```dart
maxDuration = value ?? Duration.zero
```

Reply:
fixed